File: Submissions/674_Longest_Continuous_Increasing_Subsequence.cpp
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        if(nums.empty()) return 0;
        
        int result = 1;
        int increasing_length = 1;
        for(int i=1; i<nums.size(); ++i){
            if(nums[i]-nums[i-1]>0){
                ++increasing_length;
            } else {
                result = max(result, increasing_length);
                increasing_length = 1;
            }
        }
        return max(result, increasing_length);
    }
};

File: Kotlin/674_Longest_Continuous_Increasing_Subsequence.kt
class Solution {
    fun findLengthOfLCIS(nums: IntArray): Int {
        if(nums.isEmpty()) return 0
        
        var result = 1
        var length = 1
        
        for(i in 1 until nums.size) {
            if(nums[i] - nums[i-1] > 0) {
                ++length
            } else {
                result = Math.max(result, length)
                length = 1
            }
        }
        
        return Math.max(result, length)
    }
}